systems/feide/validator.js

```diff
@@ -0,0 +1,196 @@
+    if (!systemData.uid) return error('UID mangler 🤭', systemData)
+    const data = {
+      uid: systemData.uid
+    }
+    if (systemData.uid.length === 1) return success('UID er angitt', data)
+    else if (systemData.uid.length > 1) return error('UID skal bare inneholde ett brukernavn', data)
+    else return error('UID er ikke angitt', data)
+  }),
+  test('feide-10', 'UID er likt brukernavn', 'Sjekker at UID er likt brukernavn', () => {
+    if (!systemData.uid) return error('UID mangler 🤭', systemData)
+    const data = {
+      uid: systemData.uid,
+      name: systemData.name
+    }
+    if (systemData.uid.length === 1 && systemData.uid[0] === systemData.name) return success('UID er likt brukernavn', data)
+    else if (systemData.uid.length > 1) return error('UID skal bare inneholde ett brukernavn', data)
+    else return error('UID er ikke angitt', data)
+  }),
+  test('feide-11', 'PrincipalName er satt', 'Sjekker at PrincipalName er satt', () => {
+    if (!systemData.eduPersonPrincipalName) return error('PrincipalName mangler 🤭', systemData)
+    const data = {
+      eduPersonPrincipalName: systemData.eduPersonPrincipalName
+    }
+    return success('PrincipalName er satt', data)
+  }),
+  test('feide-12', `PrincipalName er lik "uid${SYSTEMS.FEIDE.PRINCIPAL_NAME}"`, `Sjekker at PrincipalName er lik "uid${SYSTEMS.FEIDE.PRINCIPAL_NAME}"`, () => {
+    if (!systemData.eduPersonPrincipalName === `${systemData.name}${SYSTEMS.FEIDE.PRINCIPAL_NAME}`) return error('PrincipalName er feil 🤭', systemData)
+    const data = {
+      eduPersonPrincipalName: systemData.eduPersonPrincipalName
+    }
+    return success('PrincipalName er riktig', data)
+  }),
+  test('feide-13', 'E-postadresse er lik UPN', 'Sjekker at e-postadresse er lik UPN', () => {
+    if (!allData) return noData('Venter på data...')
+    if (!allData.ad) return error('Mangler AD-data', allData)
+
+    const data = {
+      feide: {
+        mail: systemData.mail
+      },
+      ad: {
+        userPrincipalName: allData.ad.userPrincipalName
+      }
+    }
+    if (systemData.mail === allData.ad.userPrincipalName) return success('E-postadresse er lik UPN', data)
+    else return error('E-postadresse er ikke lik UPN', data)
+  }),
+  test('feide-14', 'Har knyttning til en skole', 'Sjekker at det finnes knyttning til minst èn skole', () => {
+    if (!systemData.eduPersonOrgUnitDN) return error('Knyttning til skole mangler 🤭', systemData)
+    const data = {
+      eduPersonOrgUnitDN: systemData.eduPersonOrgUnitDN
+    }
+    if (systemData.eduPersonOrgUnitDN.length > 0) return success('Knyttning til skole funnet', data)
+    else return warn('Ingen knyttning til skole funnet. Dersom dette er en manuelt opprettet FEIDE-bruker eller en administrativ ansatt, er dette korrekt', data)
+  }),
+  test('feide-15', 'Har satt opp MFA', 'Sjekker at MFA er satt opp', () => {
+    if (!systemData.norEduPersonAuthnMethod) return error('MFA mangler 🤭', systemData)
```

Comment:
MFA er ikke påkrevd for elever, så bør kanskje ikke daue "hardt" her/kaste `error`?